### PR TITLE
Homework L3 arrays 1-14 + L02 updates to task 12 and 16 

### PR DESCRIPTION
**Pull request by dacian-cioc**

### CODE REVIEW

File: src/L02/homework/Task12.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import java.util.Scanner;
+
+public class Task12 {
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+
+        System.out.println("Three-digit numbers with distinct digits:");
+
+        for (int number = 100; number <= 999; number++) {
```
Comment (KaloyanTsvetkov) [inline src/L02/homework/Task12.java]: i = 102 (the first number without same digits)
i <= 987 (987 is the last number with different digits)

File: src/L02/homework/Task16.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        } else {
+            System.out.println(N + " is not a palindrome.");
+        }
+
+        scanner.close();
+    }
+
+    // Helper function to check if a number is a palindrome
+    private static boolean isPalindrome(int number) {
+        String str = Integer.toString(number);
```
Comment (KaloyanTsvetkov) [inline src/L02/homework/Task16.java]: Try to do it whitout using strings.

File: src/L02/homework/Task12.java (hunk 1 above)
Comment (dacian-cioc) [reply src/L02/homework/Task12.java]: updated in new commit https://github.com/dacian-cioc/JAVA_CORE_FOR_TA/pull/4/commits/ef6e26c49a036108145435c33dc608a4522359b1

File: src/L02/homework/Task16.java (hunk 2 above)
Comment (dacian-cioc) [reply src/L02/homework/Task16.java]: updated in new commit https://github.com/dacian-cioc/JAVA_CORE_FOR_TA/pull/4/commits/ef6e26c49a036108145435c33dc608a4522359b1

File: src/L03/homework/Task1.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+package L03.homework;
+
+import java.util.Scanner;
+
+public class Task1 {
+    public static void main(String[] args) {
+        Scanner scanner = new Scanner(System.in);
+
+        // Read the array from the user with error validation
+        int[] array = readIntArray(scanner);
```
Comment (KaloyanTsvetkov) [inline src/L03/homework/Task1.java:18]: Actually you don't need user input for this task.

File: src/L03/homework/Task6.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        String input1 = scanner.nextLine();
+        int[] array1 = convertInputToArray(input1);
+
+        // Read the second array from the user
+        System.out.print("Enter the elements of the second array (comma-separated): ");
+        String input2 = scanner.nextLine();
+        int[] array2 = convertInputToArray(input2);
+
+        // Compare the arrays
+        boolean areEqual = Arrays.equals(array1, array2);
```
Comment (KaloyanTsvetkov) [inline src/L03/homework/Task6.java:24]: Can you do it whiout using Arrays.equals

**Review by KaloyanTsvetkov — APPROVED** [review]

**Pull request merged**